pages/trouw-opties/[slug].tsx

```diff
@@ -25,68 +28,116 @@ import {
+import { MarriageOptionsContext } from "../../src/context/MarriageOptionsContext";
+import { CeremonyType } from "../../src/data/huwelijksplanner-state";
+import { resolveEmbedded } from "../../src/embedded";
+import { AvailabilitycheckService, SDGProduct, SdgproductService } from "../../src/generated";
 
 export const getServerSideProps = async ({ locale }: { locale: string }) => ({
   props: {
     ...(await serverSideTranslations(locale, ["common", "huwelijksplanner-step-2"])),
   },
 });
 
-const BlogPost: NextPage = () => {
-  const { locale = "nl", push } = useRouter();
-  const { t } = useTranslation(["common", "huwelijksplanner-step-2"]);
+type CalendarData = {
+  start: Date;
+  end: Date;
+  selectedDate?: Date;
+};
 
-  const getEvents = (type: CeremonyType, date: string) => {
-    return calendars[type as keyof typeof calendars].filter((event) => event.startDateTime.startsWith(date));
-  };
+type CeremonyData = {
+  type: CeremonyType;
+  id: string;
+  locationId: string;
+  ambtenaarId: string;
+};
 
-  const [, setResults] = useState<Availability[]>([]);
+type AvailabilitySlot = {
+  resources: string[];
+  start: string;
+  stop: string;
+};
 
-  const [selectedDate, setSelectedDate] = useState("2021-04-14");
-  const [selectedLocationAndDate, setSelectedLocationAndDate] = useState<string | undefined>();
+type Availabilities = {
+  [key: string]: Array<AvailabilitySlot>;
+};
 
-  const loadEvents = useCallback(() => {
-    const today = new Date().toISOString().replace(/T.+/, "");
-    HuwelijksplannerAPI.getAvailability({
-      interval: "PT1H",
-      start: today,
-    }).then((data) => {
-      setResults(data);
-    });
-  }, []);
+type Event = {
+  date: string;
+  emphasis?: boolean;
+  selected?: boolean;
+  disabled?: boolean;
+};
+
+const dateFormat = "yyyy-MM-dd";
+
+const PlanningFormPage: NextPage = () => {
+  const { locale = "nl", replace } = useRouter();
+  const { t } = useTranslation(["common", "huwelijksplanner-step-2"]);
+  const [marriageOptions, setMarriageOptions] = useContext(MarriageOptionsContext);
+  const [ceremonies, setCeremonies] = useState<CeremonyData[]>([]);
+  const [availabilities, setAvailabilities] = useState<Availabilities>({});
+  const [calendarData, setCalendarData] = useState<CalendarData>({
+    start: startOfMonth(Date.now()),
+    end: endOfMonth(Date.now()),
+    selectedDate: undefined,
+  });
+  const [unavailableData, setUnavailableData] = useState<Event[]>([]);
 
   useEffect(() => {
-    loadEvents();
-  }, [selectedDate, loadEvents]);
+    if (ceremonies.length === 0) return;
+    AvailabilitycheckService.availabilitycheckGetCollection({
+      resourcesCould: ceremonies.map((ceremony) => ceremony.id),
+      interval: "PT2H",
+      start: format(calendarData.start, dateFormat),
+      stop: format(calendarData.end, dateFormat),
+    }).then((response) => {
+      const availabilityResults: Availabilities = response as any;
```

Comment:
When we make custom hooks for API calls, we might want to make sure to cast the response to the correct Type so we don't have to do `as any` anymore.